docs/whatsnew/3.1.rst

```diff
@@ -24,6 +36,459 @@ smaller improvements and bug fixes, which are described in the
 * xxx pull requests have been merged since v3.0
 * xxx distinct people have contributed code
 
+.. _whatsnew-3.1-tips:
+
+Performance Tips
+================
+
+We now include a standard section called **Performance tips** in the
+documentation for each sub-package (when appropriate) which helps you get the
+best performance out of a given sub-package. For example, if you want to
+improve the performance of your code when using :class:`~astropy.units.Unit`
+and :class:`~astropy.units.Quantity`, you can read up about it in the
+:ref:`Performance tips for astropy.units <astropy-units-performance>`.
+
+.. _whatsnew-3.1-uncertainty:
+
+New Sub-package for Uncertainties and Distributions
+===================================================
+
+Astropy 3.1 includes a new experimentation sub-package: :ref:`astropy-uncertainty`.
+Currently this sub-package provides uncertainty propagation and classes for
+representing distributions using  a Monte Carlo sampling technique, which is
+useful for some cases but certainly not all.  The longer-term
+plan is to integrate other uncertainty propagation techniques to address typical
+astronomy uses cases, including integrating with the already-existing
+uncertainty classes that are focused on representing uncertainties in
+`astropy.nddata`.
+
+To demonstrate this new functionality, the example below shows propagation of
+uncertainty in the geometric mean of three numbers that have units:
+
+.. plot::
+  :include-source:
+  :align: center
+
+  import numpy as np
+  from astropy import units as u
+  from astropy import uncertainty as unc
+  from astropy.visualization import quantity_support
+  from matplotlib import pyplot as plt
+
+  np.random.seed(12345)
+
+  a = unc.normal(1.5*u.kpc, std=50*u.pc, n_samples=10000)
+  b = unc.uniform(center=3*u.kpc, width=800*u.pc, n_samples=10000)
+  c = unc.Distribution(((np.random.beta(2,5, 10000)-(2/7))/2 + 3)*u.kpc)
+  d = (a * b * c) ** (1/3)
+
+  with quantity_support():
+      plt.hist(d.distribution, bins=50)
+      plt.title(r'$\mu={0.value:.2f}$ {0.unit}, $\sigma={1.value:.2f}$ {1.unit}'.format(d.pdf_mean, d.pdf_std))
+
+
+This sub-package should be considered experimental and subject to API changes in
+the future if user feedback calls for it.
+
+
+.. _whatsnew-3.1-transit-periodogram:
+
+New Box Least Squares Periodogram
+=================================
+
+Astropy now has an implementation of the box least squares (BLS) periodogram
+that is commonly used to detect transiting exoplanets and eclipsing binary star
+systems. The interface has been designed to match the
+`~astropy.stats.LombScargle` periodogram, and it can be used with a time series
+dataset ``time``, ``flux``, and ``flux_err`` as follows::
+
+  >>> from astropy import units as u
+  >>> from astropy.stats import BoxLeastSquares
+  >>> model = BoxLeastSquares(time * u.day, flux, flux_err=0.01)  # doctest: +SKIP
```

Comment:
I'm not sure how I feel about these doctest skips, maybe don't provide this snippet as it doesn't make sense copy pasted? Whoever is interested can check out the details in the docs?

Reply:
Well the issue is generating the fake data - we don't want the example to be too long...

Reply:
yes, I got that. But the just skip the snippet, have the figures and people can go to the proper docs where there are detailed examples. I think it's worse when the copy pastes are not working than having no example snippet.

Reply:
Oh, actually it would be nice to link to the docs as it's done with the other sections (I can do this).

Reply:
done in #8109